pretixdroid/app/src/main/java/eu/pretix/pretixdroid/ui/EventinfoActivity.java

```diff
@@ -0,0 +1,338 @@
+            try {
+                return EventinfoActivity.this.api.status();
+            } catch (ApiException e) {
+                e.printStackTrace();
+            }
+            return null;
+        }
+
+        /**
+         * it parses the answer of the pretix endpoint into objects and adds them to the list
+         * @param result the answer of the pretix status endpoint
+         */
+        @Override
+        protected void onPostExecute(JSONObject result) {
+            EventItemAdapter eia = EventinfoActivity.this.mAdapter;
+            try {
+                EventCardItem ici = new EventCardItem(result);
+                eia.addItem(ici);
+
+                JSONArray items = result.getJSONArray("items");
+                for (int i = 0; i < items.length(); i++) {
+                    EventItemCardItem eici = new EventItemCardItem(items.getJSONObject(i));
+                    eia.addItem(eici);
+                }
+            } catch (JSONException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    /**
+     * implementation of an adapter for a listview to hold EventCards and EventItemCards
+     */
+    public class EventItemAdapter extends BaseAdapter {
+        public static final int TYPE_EVENTCARD = 0;
+        public static final int TYPE_EVENTITEMCARD = 1;
+        private static final int MAX_TYPES = 2;
+
+        private final ArrayList<EventinfoListItem> mData = new ArrayList<>();
+        public final LayoutInflater mInflater;
+
+        public EventItemAdapter(Context ctx) {
+            mInflater = LayoutInflater.from(ctx);
+        }
+
+        public void addItem(EventinfoListItem item) {
+            this.mData.add(item);
+            notifyDataSetChanged();
+        }
+
+        @Override
+        public int getItemViewType(int position) {
+            return this.mData.get(position).getType();
+        }
+
+        @Override
+        public int getViewTypeCount() {
+            return MAX_TYPES;
+        }
+
+        @Override
+        public int getCount() {
+            return mData.size();
+        }
+
+        @Override
+        public EventinfoListItem getItem(int position) {
+            return mData.get(position);
+        }
+
+        @Override
+        public long getItemId(int position) {
+            return position;
+        }
+
+        @Override
+        public View getView(int position, View convertView, final ViewGroup parent) {
+            if (convertView == null) {
+                EventinfoListItem item = this.getItem(position);
+                return item.getCard(mInflater, parent);
+            } else {
+                ((EventinfoListItem) convertView.getTag()).fillView(convertView, mInflater, parent);
+                return convertView;
+            }
+
+        }
+
+    }
+
+    /**
+     * interface of all CardItems that are displayed on the eventinfo page
+     */
+    public interface EventinfoListItem {
```

Comment:
This file gets really large which will be hard to manage over time. Could you refactor at least the interfaces into their own files? I see the use of having the adapter classes nested for shared property acces, but that is not required here.